contrib/dracut/90zfs/zfs-lib.sh.in

```diff
@@ -74,14 +74,23 @@ import_pool() {
 # mount_dataset DATASET
 #   mounts the given zfs dataset.
 mount_dataset() {
-        dataset="${1}"
+    dataset="${1}"
     mountpoint="$(zfs get -H -o value mountpoint "${dataset}")"
-
+    mountopts="atime"
+    RPOOL=$(echo "${dataset}" | awk -F\/ '{print $1}')
+    ATIME_OPTION=$(zfs get -H -o value -s default,local,inherited atime $(zpool get -H -o value bootfs $RPOOL))
```

Comment:
I don't understand why you are going to this work to go from a dataset to the pool and then the bootfs dataset. I also don't understand the `-s default,local,inherited` part. Why isn't this just: ATIME_OPTION=$(zfs get -H -o value atime "${dataset}")

Reply:
because it can be temporary incorrect value, this way we only get the explicitly set or default option.

I think the other part I just overthought and didn't fix when I copied from my test scripts.